src/router/router.js

```diff
@@ -129,14 +185,34 @@ export const matchHash = (path, routes = []) => {
   return matchingRoute
 }
 
+/**
+ * Navigate to a route
+ *
+ * This isn't the prettiest way to do this, but it works. The reason is that extends
+ * only works for Classes or Factory functions. As such we need to use this
+ * @typedef {BlitsComponent & {
+ *   activeView: BlitsComponent
+ * }} RouterViewComponent
+ *
+ * @this {RouterViewComponent} this
+ * @returns {Promise<void>}
+ */
 export const navigate = async function () {
   Announcer.stop()
   Announcer.clear()
   state.navigating = true
   if (this.parent[symbols.routes]) {
     let previousRoute = currentRoute //? Object.assign({}, currentRoute) : undefined
     const { hash, path, queryParams } = getHash()
-    let route = matchHash(path, this.parent[symbols.routes])
+    const routeOrFalse = matchHash(path, this.parent[symbols.routes])
+
+    if (!routeOrFalse) {
+      // If no route is found, probably handle with a generic route?
+      return
+    }
+
+    /** @typedef {Route} */
+    let route = routeOrFalse
```

Comment:
can we remove this code change from this PR? There is an issue open regarding error handling when route doesn't exist, and a feature underway to handle this case gracefully